### PR TITLE
samples: nrf_desktop: Rework configuration

### PR DESCRIPTION
**Pull request by pdunaj**

### CODE REVIEW

File: samples/nrf_desktop/configuration/nrf52840_pca10059/led_state_def.h [hunk 1]
```diff
@@ -6,20 +6,16 @@
...
-	},
-	{
-		DT_NORDIC_NRF_PWM_PWM_1_CH0_PIN,
-	}
-};
+/* This file must be included only once */
+const struct {} led_state_def_include_once;
 
-const u8_t led_map[LED_ID_COUNT] = {0, 1};
+/* Map function to LED id */
```
Comment (MarekPieta) [inline samples/nrf_desktop/configuration/nrf52840_pca10059/led_state_def.h]: ID ?

File: samples/nrf_desktop/configuration/nrf52840_pca20041/led_state_def.h [hunk 2]
```diff
@@ -6,24 +6,16 @@
...
-		DT_NORDIC_NRF_PWM_PWM_1_CH0_PIN,
-		DT_NORDIC_NRF_PWM_PWM_1_CH1_PIN,
-		DT_NORDIC_NRF_PWM_PWM_1_CH2_PIN
-	}
-};
+/* This file must be included only once */
+const struct {} led_state_def_include_once;
 
-const u8_t led_map[LED_ID_COUNT] = {0, 1};
+/* Map function to LED id */
```
Comment (MarekPieta) [inline samples/nrf_desktop/configuration/nrf52840_pca20041/led_state_def.h]: ID ?

File: samples/nrf_desktop/src/modules/led_state.c [hunk 3]
```diff
@@ -155,6 +156,8 @@ static bool event_handler(const struct event_header *eh)
 		struct module_state_event *event = cast_module_state_event(eh);
 
 		if (check_state(event, MODULE_ID(main), MODULE_STATE_READY)) {
+			static_assert(LED_ID_COUNT <= CONFIG_DESKTOP_LED_COUNT,
+				      "Not enough LEDS configured");
```
Comment (MarekPieta) [inline samples/nrf_desktop/src/modules/led_state.c]: LEDs

File: samples/nrf_desktop/configuration/nrf52840_pca10059/led_state_def.h (hunk 1 above)
Comment (pdunaj) [reply samples/nrf_desktop/configuration/nrf52840_pca10059/led_state_def.h]: fix

File: samples/nrf_desktop/configuration/nrf52840_pca20041/led_state_def.h (hunk 2 above)
Comment (pdunaj) [reply samples/nrf_desktop/configuration/nrf52840_pca20041/led_state_def.h]: fixed

File: samples/nrf_desktop/src/modules/led_state.c (hunk 3 above)
Comment (pdunaj) [reply samples/nrf_desktop/src/modules/led_state.c]: fixed

**Comment by pdunaj** [comment]
hi @pizi-nordic , could you merge?

**Pull request merged**